framework/src/main/java/org/checkerframework/framework/util/ContractsUtils.java

```diff
@@ -413,21 +414,83 @@ private AnnotationMirror getAnnotationMirrorOfMetaAnnotation(
 
         // Check type-system specific annotations.
         Class<PostconditionAnnotation> metaAnnotation = PostconditionAnnotation.class;
+        Class<String> stringType = String.class;
+        Class<AnnotationMirror> annotationmirrorType = AnnotationMirror.class;
         List<Pair<AnnotationMirror, AnnotationMirror>> declAnnotations =
                 factory.getDeclAnnotationWithMetaAnnotation(methodElement, metaAnnotation);
         for (Pair<AnnotationMirror, AnnotationMirror> r : declAnnotations) {
             AnnotationMirror anno = r.first;
             AnnotationMirror metaAnno = r.second;
-            List<String> expressions =
-                    AnnotationUtils.getElementValueArray(anno, "value", String.class, false);
-            AnnotationMirror postcondAnno = getAnnotationMirrorOfMetaAnnotation(metaAnno, anno);
+
+            result.addAll(
+                    getPostConditionAnnotations(anno, metaAnno, annotationmirrorType, stringType));
+        }
+
+        return result;
+    }
+
+    /**
+     * Returns postcondition annotations.
+     *
+     * @param anno the type qualifier that {@code methodElement} should have
+     * @param metaAnno postCondition meta-annotation
+     * @param annotationmirrorType the annotationMirror class type used to cast the element value
+     * @param stringType the string class type used to cast the element value
+     */
+    private Set<Postcondition> getPostConditionAnnotations(
+            AnnotationMirror anno,
+            AnnotationMirror metaAnno,
+            Class<AnnotationMirror> annotationmirrorType,
+            Class<String> stringType) {
+
+        @SuppressWarnings("unchecked")
+        List<AnnotationValue> la =
+                AnnotationUtils.getElementValue(anno, "value", List.class, false);
+
+        Set<Postcondition> result = new LinkedHashSet<>();
+
+        // Check for multiple contracts.
+        List<AnnotationMirror> annotations = new ArrayList<>(la.size());
+        for (AnnotationValue a : la) {
+            if (a.getClass().getName().equals("javax.lang.model.element.AnnotationMirror")
+                    || a.getClass()
+                            .getName()
+                            .equals("com.sun.tools.javac.code.Attribute$Compound")) {
+                annotations.add(annotationmirrorType.cast(a.getValue()));
```

Comment:
I don't see a reason to pass `annotationmirrorType` in as a parameter. Can you explain?

Reply:
Yes, we can define it in the function too. I have updated it. Please review it.